components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/DAOIntegrationTestBase.java

```diff
@@ -38,39 +42,71 @@ public DAOIntegrationTestBase() {
         }
     }
 
+    @BeforeClass
+    public void init() throws Exception {
+        // This used to check connection healthy
+        if ("h2".equals(database)) {
+            dataSource = new H2DataSource();
+        } else if ("mysql".contains(database)) {
+            dataSource = new MySQLDataSource();
+        } else if ("postgres".contains(database)) {
+            dataSource = new PostgreDataSource();
+        } else if ("mssql".contains(database)) {
+            dataSource = new MSSQLDataSource();
+        } else if ("oracle".contains(database)) {
+            dataSource = new OracleDataSource();
+        }
+        int maxRetries = 5;
+        long maxWait = 5000;
+        while (maxRetries > 0) {
+            try {
+                dataSource.getConnection();
+                log.warn("Database Connection Successful");
+                break;
+            } catch (Exception e) {
```

Comment:
This throws SqlException but when connectivity issue it throws generic exception